templates/problem/voting-controls.html

```diff
@@ -0,0 +1,211 @@
+        font-size: 24px;
+        font-weight: bold;
+        padding:10px
+    }
+
+    .vote-form-value {
+        font-size: 24px;
+    }
+
+    .vote-form-text {
+        font-size: 18px;
+    }
+
+    .voting-form-error {
+        font-size: 18px;
+        color: red;
+    }
+
+    .vote-button {
+        float:right;
+    }
+
+    .form-button {
+        cursor: pointer;
+    }
+</style>
+
+{% if not in_contest %}
+    <hr style="padding-top: 0.1em">
+{% endif %}
+
+{% include 'lightbox.html' %}
+
+{% if can_vote %}
+    <div class="vote-form" id="id_vote_form_box">
+        <div id="id_has_voted_prompt">
+            <span class="vote-form-info">{{ _('Current Vote:') }}</span>
+            <span id="id_current_vote_value" class="vote-form-value"></span>
+            <br/>
+            <br/>
+            <span class="vote-form-text">{{  _('Change your vote below!') }}</span>
+            <br/>
+            <br/>
+        </div>
+        <div id="id_has_not_voted_prompt">
+            <span class="vote-form-text">{{ _("Cast your vote on this problem's weightage below!") }}</span>
+            <br/>
+            <br/>
+        </div>
+        <form id="id_vote_form">
+            {% csrf_token %}
+            <span class="vote-form-info">{{ _('Points') }}</span>
+            <input id="id_points" class="vote-form-value" type="number" step="any" name="points">
+            <script>
+                $('#id_points').keydown(e => {
+                    const invalidChars = ['e','+','-','.'];
+                    if (invalidChars.includes(e.key)) {
+                        e.preventDefault();
+                    }
+                });
+            </script>
+            <br/>
+            <div id="id_points_error_box">
+                <br/>
+                <span id="id_points_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <span class="vote-form-info">{{ _('Note') }}</span>
+            <br/>
+            <textarea id="id_note" class="vote-form-text" name="note" rows="6" cols="80"></textarea>
+            <br/>
+            <div id="id_note_error_box">
+                <br/>
+                <span id="id_note_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <input type="submit" id="id_vote_form_submit_button" class="vote-button" value="{{ _('Vote') }}">
+            <input type="hidden" name="vote_confirmation"/>
+        </form>
+    </div>
+
+    <a class="form-button" id="id_vote_button"></a>
+
+    <form id="id_delete_vote_form">
+        {% csrf_token %}
+        <a class="form-button" id="id_delete_vote_button">{{ _('Delete vote') }}</a>
+    </form>
+
+    <script>
+        let voted_points = null;
+        let voted_note = null;
+        {% if has_voted %}
+            let has_voted = true;
+            voted_points = {{ voted_points }};
+            voted_note = `{{ voted_note }}`;
```

Comment:
I would highly discourage using templating in script tags since Jinja is not meant to be used in JS templating. This one in particular is an easy XSS attack with text interpolation: `${alert(1)}`. If you want to use data, either parse it from the DOM (`data-` attributes) or use XHR to fetch it.

Reply:
Though I overall agree, would `"{{ voted_note|escape }}"` be safe here?

Reply:
Why can't you just use double quotes instead of backtics?

Reply:
I removed the backticks in my fix.

Reply:
Actually, can you look into https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#json-script?

Reply:
Fixed.